python/fusion_engine_client/analysis/data_loader.py

```diff
@@ -2,13 +2,16 @@
 
 from collections import deque
 import copy
-from datetime import datetime
+from datetime import datetime, timezone
 import io
 import os
 
+from gpstime import gpstime, unix2gps
 import numpy as np
+import scipy as sp
```

Comment:
FYI: it doesn't seem like there's a universal way to abbreviate "scipy", although I guess `sp` is the most-used abbreviation: https://import-as.github.io/import/scipy.html

It seems like in the scipy documentation, it doesn't really ever use a `import scipy` statement but rather uses `from scipy import ____`. Doesn't matter too much.

Reply:
Yeah, for scipy I've most commonly seen either `import scipy as sp` (this way) and `from scipy import x`. In general, for really big libraries like scipy and numpy where you're probably gonna call lots of different functions and classes, I prefer to use `import thing` (or `import really_long_thing as short_alias`), that way everything is scoped under the `thing` name like `thing.ClassIWant` and not just wholesale dumped into your file's namespace. For numpy, `import numpy as np` is extremely common, and for scipy `import scipy as sp` is also pretty common and consistent with numpy, which almost always is present too if you're using scipy.

I find scoping like that especially important in a file within a library that is meant for someone else to import, like this one. There, if someone wanted to do `from data_loader import *`, if we had just done `from numpy import *`, then the caller would _also_ unexpectedly get all numpy stuff suddenly in their file. With this way, they technically _will_ have `np` present in their file, which they didn't ask for, but at least it's just one thing. Still though, I personally wouldn't recommend doing `from data_loader import *` to avoid that problem altogether. I'd suggest just `from data_loader import DataLoader`.